### PR TITLE
Make gleam optional for compositor, switch various names from glutin to winit

### LINKED ISSUE #20612: Avoid direct GL calls
Servo needs to be fully abstracted away from GL by WebRender (and, to an extent, `offscreen_gl_context`). Currently I see [draw_img](https://github.com/servo/servo/blob/bfb9fe6159ba61f0155a5ecc5a473b2a3edd6ef6/components/compositing/compositor.rs#L1381-L1417) issuing direct calls.

### PR DESCRIPTION
**Pull request by kvark**
The PR goes an extra step towards discriminating GL to a level of an optional dependency. This would ease up the transition to gfx-hal when the time calls for it, and upstreams some of the patches we've been using locally, cleans up some of the old cruft along the way.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach build-geckolib` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #20612

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____ it's just refactoring

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20658)

### CODE REVIEW

**Comment by kvark** [comment]
TaskCluster things that:
> It looks like rustup is not installed. See instructions at https://github.com/servo/servo/#setting-up-your-environment

**Comment by emilio** [comment]
Yeah, please ignore taskcluster, we don't gate on it.

The huge number of unexpected passes look like the screenshotting code may have broken? I'd retry but travis is red (need to avoid unwrap() in compositing).

**Comment by kvark** [comment]
@emilio thanks for taking a look! I fixed the unwraps as well as the part of refactor affecting screenshots. :crossed_fingers: for the following CI build.

**Comment by kvark** [comment]
all is green here
@bors try

**Comment by paulrouget** [comment]
What is "the transition to gfx-hal"?

**Comment by kvark** [comment]
@emilio looks like we are reviewed and tested, all set to sail?

@paulrouget 
For the matter of this PR - it's just to pave the way for other native graphics API backends than GL.

In a greater context, WebRender is getting ported to [gfx-rs](https://github.com/gfx-rs/gfx), and we sometimes mention it as gfx-hal to distinguish the new low-level API from the old one (that is currently on crates.io).
I need servo to glue with gfx-hal nicely in order to prototype the [WebGPU](https://www.w3.org/community/gpu/) implementation.

File: components/compositing/windowing.rs [hunk 1]
```diff
@@ -6,17 +6,20 @@
...
 use script_traits::{MouseButton, TouchEventType, TouchId};
 use servo_geometry::{DeviceIndependentPixel, DeviceUintLength};
 use servo_url::ServoUrl;
 use std::fmt::{Debug, Error, Formatter};
+#[cfg(feature = "gleam")]
 use std::rc::Rc;
 use style_traits::DevicePixel;
 use webrender_api::{DeviceIntPoint, DevicePoint, DeviceUintSize, DeviceUintRect, ScrollLocation};
 
+
```
Comment (emilio) [inline components/compositing/windowing.rs]: nit: stray newline I guess, but not a big deal

File: ports/servo/glutin_app/keyutils.rs [hunk 2]
```diff
@@ -131,129 +131,132 @@ pub fn char_to_script_key(c: char) -> Option<constellation_msg::Key> {
...
+        Down => Ok(Key::Down),
 
-        VirtualKeyCode::LShift => Ok(Key::LeftShift),
-        VirtualKeyCode::LControl => Ok(Key::LeftControl),
-        VirtualKeyCode::LAlt => Ok(Key::LeftAlt),
-        VirtualKeyCode::LWin => Ok(Key::LeftSuper),
-        VirtualKeyCode::RShift => Ok(Key::RightShift),
-        VirtualKeyCode::RControl => Ok(Key::RightControl),
-        VirtualKeyCode::RAlt => Ok(Key::RightAlt),
-        VirtualKeyCode::RWin => Ok(Key::RightSuper),
```
Comment (emilio) [inline ports/servo/glutin_app/keyutils.rs]: These changes look mostly unrelated, would've been nice to make a different commit for them.

File: ports/servo/glutin_app/keyutils.rs [hunk 3]
```diff
@@ -131,129 +131,132 @@ pub fn char_to_script_key(c: char) -> Option<constellation_msg::Key> {
     }
 }
 
-pub fn glutin_key_to_script_key(key: winit::VirtualKeyCode) -> Result<constellation_msg::Key, ()> {
+pub fn winit_key_to_script_key(
```
Comment (emilio) [inline ports/servo/glutin_app/keyutils.rs]: This fits in one line right? Why the indentation change?

**Review by emilio — APPROVED** [review]
Looks good! None are blocking issues so just feel free to `r=emilio` as-is, or after addressing the comments :)

**Comment by kvark** [comment]
Another (fair) point raised by @emilio is that since we are adding a feature to `compositing` module, it would be nice to test it on CI without the feature. Strangely, I'm getting errors trying to build either `compositing` or `servo_url` with `./mach cargo -p <something>`, regardless of the features...

**Comment by kvark** [comment]
@emilio changes are addressed (except for the CI testing, given a blocker filed separately). PTAL

**Review by emilio — APPROVED** [review]
Looks good, feel free to r=me when you please :)

**Comment by kvark** [comment]
Thanks! Added those references now.
@bors-servo r=emilio

**Comment by kvark** [comment]
Weird to see it failing to fetch the sources :(
@bors-servo retry
